### ISSUE TITLE
[$250] [Dev][StrictMode] Popover menu has no background smallscreen

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#45385]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**  DEV only
**Reproducible in production?:** DEV only
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @ishpaul777 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1720819445275459
 
## Action Performed:
Long press on any report in LHN

## Expected Result:
Popover menus haves translucent background
## Actual Result:
 Popover menu dont have background
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/894fc80a-baf9-4aac-b71f-43c212d380a5

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ff564d543a9355a6</li>
        <li>Upwork Job ID: 1812925525309666872</li>
        <li>Last Price Increase: 2024-07-29</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Contributor | 103339816</li>
</ul></li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
This has been labelled "Needs Reproduction". Follow the steps here: https://stackoverflowteams.com/c/expensify/questions/16989

**Comment by lschurr** [comment]
I'm not able to test since I don't work in a DEV environment. @allroundexperts could you try reproducing this?

**Comment by allroundexperts** [comment]
It's easily reproducible @lschurr.

https://github.com/user-attachments/assets/d6b6fc22-991d-428a-97f7-43d1eda4708b

**Comment by isogit123** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Popover menu has no background in small screens in the development environment when using the strict mode.
### What is the root cause of that problem?
`ReactNativeModal` isn't showing the backdrop background and the opacity of the background is set to zero when the strict mode is enabled. The background appears as expected if the strict mode is disabled.
### What changes do you think we should make in order to solve the problem?

Using the `ReactNativeModal` ref, we can set the opacity to the desired value so the background appears as expected by adding the following code snippet:
```
const modalRef = useRef(null);
try{
    modalRef.current.backdropRef.ref.style.opacity = !shouldUseCustomBackdrop && hideBackdrop ? 0 : variables.overlayOpacity;
}
catch {}
```
in line 209 at: https://github.com/Expensify/App/blob/105dce5151053bf6c81e60e68a610dd888c5da45/src/components/Modal/BaseModal.tsx#L205-L218
then bind the ref to the `ReactNativeModal` at line 224 in the following code snippet:
https://github.com/Expensify/App/blob/105dce5151053bf6c81e60e68a610dd888c5da45/src/components/Modal/BaseModal.tsx#L220-L230
Below is a screenshot after applying the change:
<img src="https://github.com/user-attachments/assets/926945f4-f1e9-44bf-a906-303e8497a74f" width="300" />

### What alternative solutions did you explore? (Optional)
The last commit to the React Native Modal library used in the Popover menu is 2 years old and is looking for maintainers, which is mentioned in https://github.com/react-native-modal/react-native-modal?tab=readme-ov-file#announcements, so we can use the React Native built-in `Modal` component or any other wrapper library that is compatible with the strict mode and maintained.

**Comment by allroundexperts** [comment]
Thanks for the proposal @dominictb. Your RCA seems to be accurate, but I'm not sure if patching / editing the package is the correct way to proceed here. Software mansion maintains the react-native-reanimated library, and I think it would be great if we can consult about this with them in the open source channel. Can you spin up a message and tag me? Thanks!

**Comment by allroundexperts** [comment]
@isogit123 Your RCA is incomplete. Please make sure to include proper root cause in order for the proposal to be accepted.

**Comment by isogit123** [comment]
<a href="https://github.com/Expensify/App/issues/45385#issuecomment-2229674626">Proposal updated</a>

**Comment by lschurr** [comment]
Any update on this one @allroundexperts?

**Comment by allroundexperts** [comment]
> @allroundexperts sorry but I think we are referring to 2 different libraries here. The package mentioned in my proposal is `react-native-animatable`, not `react-native-reanimated`. I don't think we maintain the `react-native-animatable` right?

Yes, my bad. Thinking about it now, can we just use react-native-reanimated only here?

**Comment by lschurr** [comment]
Bump on this @allroundexperts

**Comment by allroundexperts** [comment]
@dominictb Do we have any existing patches in our codebase for this library?

**Comment by allroundexperts** [comment]
Okay. Can we somehow not use `react-native-animatable` in our codebase and just use `react-native-reanimated` instead?

**Comment by allroundexperts** [comment]
> We still need to patch `react-native-animatable` to fix the second issue mentioned here: [#45385 (comment)](https://github.com/Expensify/App/issues/45385#issuecomment-2244211012)

Ah, I see. I guess that's the best option then.

**Comment by allroundexperts** [comment]
@dominictb's [proposal](https://github.com/Expensify/App/issues/45385#issuecomment-2231387657) looks good to me. Let's create a patch for the package.

🎀 👀 🎀 C+ reviewed

**Comment by lschurr** [comment]
Any update @dominictb?

**Comment by flodnv** [comment]
Could you please review the PR @allroundexperts ?

**Comment by allroundexperts** [comment]
Reviewed!

**Comment by allroundexperts** [comment]
@dominictb Any update here on your new PR? @lschurr This is still not fixed.

**Comment by alexpensify** [comment]
Ok, noted

**Comment by alexpensify** [comment]
@dominictb any update here?

**Comment by alexpensify** [comment]
Thanks, I have removed you and we are back to looking for proposals here.

**Comment by lschurr** [comment]
Removing Al since I'm back to work - @allroundexperts - is this still broken? Are we waiting on new proposals?

**Comment by lschurr** [comment]
Bump on this one @allroundexperts - can you confirm this is bug is still happening?

**Comment by lschurr** [comment]
Bumped the original Slack thread: https://expensify.slack.com/archives/C049HHMV9SM/p1720819445275459

**Comment by lschurr** [comment]
Thank you! Going to close this one.

**Comment by allroundexperts** [comment]
@lschurr I reviewed a PR here that caused a regression. Can I get a payment summary, please?
